chia/types/eligible_coin_spends.py

```diff
@@ -24,6 +24,7 @@ class EligibilityAndAdditions:
     is_eligible_for_dedup: bool
     spend_additions: list[Coin]
     is_eligible_for_ff: bool
+    ff_puzzle_hash: Optional[bytes32] = None
```

Comment:
would you mind adding a comment here? I imagine if `is_eligible_for_ff` is `True`, this is not allowed to be `None` and vice versa.

So, technically, the `is_eligible_for_ff` could also be changed from `bool` -> `Optional[bytes32]`.

Reply:
That's reasonable.